exporters/metric/prometheus/prometheus.go

```diff
@@ -267,13 +275,13 @@ func (c *collector) exportCounter(ch chan<- prometheus.Metric, sum aggregator.Su
 func (c *collector) exportSummary(ch chan<- prometheus.Metric, dist aggregator.Distribution, kind core.NumberKind, desc *prometheus.Desc, labels []string) error {
 	count, err := dist.Count()
 	if err != nil {
-		return err
+		return fmt.Errorf("error retrieving count: %w", err)
 	}
 
 	var sum core.Number
 	sum, err = dist.Sum()
 	if err != nil {
-		return err
+		return fmt.Errorf("error retrieving sum: %w", err)
```

Comment:
I would probably be useful to include the aggregator in the error message to disabibuate the returned errors (there are two `"error retrieving sum: %w"` messages).

Maybe something like:

```
"error retrieving distribution sum: %w"
```

Reply:
The error is namespaced at the call site with the aggregator in `Collect`. (See: https://play.golang.org/p/gSk-0oS_nuJ)
But I will incorporate your suggestion. (Tomorrow, it's already late here in Europe 😅)

Reply:
updated